lib/spack/spack/build_environment.py

```diff
@@ -345,8 +345,7 @@ def set_module_variables_for_package(pkg, module):
     m.ctest = Executable('ctest')
 
     # standard CMake arguments
-    m.std_cmake_args = ['-DCMAKE_INSTALL_PREFIX=%s' % pkg.prefix,
-                        '-DCMAKE_BUILD_TYPE=RelWithDebInfo']
+    m.std_cmake_args = get_std_cmake_args(pkg)
     if platform.mac_ver()[0]:
         m.std_cmake_args.append('-DCMAKE_FIND_FRAMEWORK=LAST')
```

Comment:
Yes thanks, good catch.  Did it.

Now I remember the logic behind this code: I needed `std_cmake_args` from a
point where they were not available from the results of
`set_module_variables_for_package()`.  So I took the code out and put it in
a separate function, called from `set_module_variables_for_package()` (and
elsewhere where I needed it).  Merge issues at some point caused the old
code to sneak back in.

I think we now have what was intended.

On Mon, Jul 18, 2016 at 9:57 AM, Massimiliano Culpo <
notifications@github.com> wrote:

> In lib/spack/spack/build_environment.py:
>
> > @@ -345,8 +345,7 @@ def set_module_variables_for_package(pkg, module):
> >      m.ctest = Executable('ctest')
> >
> >      # standard CMake arguments
> > -    m.std_cmake_args = ['-DCMAKE_INSTALL_PREFIX=%s' % pkg.prefix,
> > -                        '-DCMAKE_BUILD_TYPE=RelWithDebInfo']
> > +    m.std_cmake_args = get_std_cmake_args(pkg)
> >      if platform.mac_ver()[0]:
> >          m.std_cmake_args.append('-DCMAKE_FIND_FRAMEWORK=LAST')
>
> @citibeth <https://github.com/citibeth> Shouldn't this and the following
> few lines be deleted too ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LLNL/spack/pull/1250/files/94e3d271d6889880333e4ca9bb89f2e2defda489#r71151069>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB1cd1IevSJNfnfg0iOVRfaIQ4VJjdJXks5qW4YzgaJpZM4JMZOj>
> .
>